hal/src/b5som/network/quectel_ncp_client.cpp

```diff
@@ -155,48 +155,84 @@ int QuectelNcpClient::initParser(Stream* stream) {
+                                    // Parse response ignoring mode (replicate URC response)
+                                    int r = ::sscanf(atResponse, "+CREG: %*d,%d,\"%x\",\"%x\",%d", &val[0], &val[1], &val[2], &val[3]);
+                                    // Reparse URC as direct response
+                                    if (0 >= r) {
+                                        r = CHECK_PARSER_URC(
+                                            ::sscanf(atResponse, "+CREG: %d,\"%x\",\"%x\",%d", &val[0], &val[1], &val[2], &val[3]));
+                                    }
+                                    CHECK_TRUE(r >= 1, SYSTEM_ERROR_AT_RESPONSE_UNEXPECTED);
                                     // Home network or roaming
-                                    if (val[r - 1] == 1 || val[r - 1] == 5) {
+                                    if (val[0] == 1 || val[0] == 5) {
                                         self->creg_ = RegistrationState::Registered;
                                     } else {
                                         self->creg_ = RegistrationState::NotRegistered;
                                     }
                                     self->checkRegistrationState();
+                                    // Cellular Global Identity (partial)
+                                    self->cgi_.location_area_code = static_cast<uint16_t>(val[1]);
+                                    self->cgi_.cell_id = val[2];
                                     return SYSTEM_ERROR_NONE;
                                 },
                                 this));
     CHECK(parser_.addUrcHandler("+CGREG",
                                 [](AtResponseReader* reader, const char* prefix, void* data) -> int {
                                     const auto self = (QuectelNcpClient*)data;
-                                    int val[2];
-                                    int r = CHECK_PARSER_URC(reader->scanf("+CGREG: %d,%d", &val[0], &val[1]));
-                                    CHECK_TRUE(r >= 1, SYSTEM_ERROR_UNKNOWN);
+                                    int val[4] = {-1,-1,-1,-1};
+                                    char atResponse[64] = {0};
+                                    // Take a copy of AT response for multi-pass scanning
+                                    CHECK_PARSER_URC(reader->readLine(atResponse, sizeof(atResponse)));
+                                    // Parse response ignoring mode (replicate URC response)
+                                    int r = ::sscanf(atResponse, "+CGREG: %*d,%d,\"%x\",\"%x\",%d,\"%*x\"", &val[0], &val[1], &val[2], &val[3]);
+                                    // Reparse URC as direct response
+                                    if (0 >= r) {
+                                        r = CHECK_PARSER_URC(
+                                            ::sscanf(atResponse, "+CGREG: %d,\"%x\",\"%x\",%d,\"%*x\"", &val[0], &val[1], &val[2], &val[3]));
```

Comment:
Nit: Why 5 parameters here, while this line (https://github.com/particle-iot/device-os/blob/82486ffdf7811904faa3dbc1ecad2e52377ecc49/hal/src/b5som/network/quectel_ncp_client.cpp#L223) has only 4 ?